vector/src/main/java/im/vector/app/features/pin/lockscreen/pincode/PinCodeHelper.kt

```diff
@@ -36,16 +40,27 @@ class PinCodeHelper @Inject constructor(
      * Creates a PIN code key if needed and stores the PIN code encrypted with it.
      */
     suspend fun createPinCode(pinCode: String) {
-        val encryptedValue = lockScreenKeyRepository.encryptPinCode(pinCode)
+        val encryptedValue = lockScreenKeyRepository.encryptPinCode(pinCode.sha256())
         encryptedStorage.savePinCode(encryptedValue)
+
+        sharedPreferences.edit {
+            putBoolean(IS_HASHED_PIN_CODE, true)
+        }
     }
 
     /**
      * Verifies the passed [pinCode] against the encrypted one.
      */
     suspend fun verifyPinCode(pinCode: String): Boolean {
         val encryptedPinCode = encryptedStorage.getPinCode() ?: return false
-        return lockScreenKeyRepository.decryptPinCode(encryptedPinCode) == pinCode
+        val decryptedPinCode = lockScreenKeyRepository.decryptPinCode(encryptedPinCode)
+
+        return if (!sharedPreferences.getBoolean(IS_HASHED_PIN_CODE, false)) {
```

Comment:
I would invert if and else to avoid using `!`

Reply:
Done